src/App.tsx

```diff
@@ -1,14 +1,67 @@
+/* eslint-disable no-alert */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
+import { Filter } from './enum/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [userId, setUserId] = useState(0);
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
```

Comment:
can we store selectedTodoId here? number | null
to avoid passing undefined to TodoList component